esp/bindings/SOAP/Platform/soapparam.hpp

```diff
@@ -1022,6 +1022,8 @@ class SoapEnumParamNew
 
     void marshall(IRpcMessage &rpc_call, const char *tagname, const char* itemname, const char* elementtype="",const char *basepath="", const char *prefix="")
     {
+        if ((value < 0) || (value >= enumstrings.length()))
+            value = (enumtype) 0;
```

Comment:
Is value = 0 considered the default?  What happens if we don't add_value at all?

Reply:
I made the change. If we don't add_value at all, it does create an empty item which is we want.